framework/src/multiapps/TransientMultiApp.C

```diff
@@ -314,7 +311,7 @@ TransientMultiApp::solveStep(Real dt, Real target_time, bool auto_advance)
       {
         _console << "Solving Normal Step!" << std::endl;
 
-        if (_first)
+        if (_first && !_app.isRecovering())
```

Comment:
We have to `advanceState()` every time step. However, this particular call to `advanceState()` is in the setup of a Multiapp which is only called once per simulation. On recover we still run this routine but we don't want to advance state again since it's done as part of the normal execution cycle.

This whole advance state stuff _before_ the first solve is screwy and should be removed but you can read more about why it's necessary in the comment and logic here:
https://github.com/idaholab/moose/blob/devel/framework/src/executioners/Transient.C#L231-L236